### PR TITLE
Consensus const change

### PR DESCRIPTION
**Pull request by iulianpascalau**
## Description of the reasoning behind the pull request
- during the operations of the backup nodes, there may be situations in which a backup node can not "see" the main node's messages because the rest of the network will actively drop the second consensus message. This will make the backup node to send more consensus messages until a message from the master machine is received.

## Proposed Changes
- make the consensus accept a maximum of 2 messages per public key, not just 1 as it was before

## Testing procedure
- standard system test
- a test without transactions, all nodes started with the following debug log level: `*:DEBUG,consensus:TRACE,p2p:TRACE,debug:DEBUG`. Stop one validator from shard X, replace its validatorKey.pem with another validator from shard X and then start it. We should be able to check in the logs that all nodes from shard X received at least 2 messages from that public key

### CODE REVIEW

File: consensus/spos/constants.go [hunk 1]
```diff
@@ -23,4 +23,4 @@ const ValidatorPeerHonestyDecreaseFactor = -2
 
 // MaxNumOfMessageTypeAccepted represents the maximum number of the same message type accepted in one round to be
 // received from the same public key
-const MaxNumOfMessageTypeAccepted = 1
+const MaxNumOfMessageTypeAccepted = 2
```
Comment (AdoAdoAdo) [inline consensus/spos/constants.go:26]: can we change the num of message types only for signatures?
not for the other messages

File: consensus/spos/consensusMessageValidator_test.go [hunk 2]
```diff
@@ -533,8 +533,15 @@ func TestCheckConsensusMessageValidity_ErrMessageTypeLimitReached(t *testing.T)
 		ChainID: chainID, MsgType: int64(bls.MtBlockBodyAndHeader),
 		Header: headerBytes, BlockHeaderHash: headerHash, PubKey: pubKey, Signature: sig, RoundIndex: 10,
 	}
-	err := cmv.CheckConsensusMessageValidity(cnsMsg, "")
-	assert.True(t, errors.Is(err, spos.ErrMessageTypeLimitReached))
+	t.Run("second time the message is received should still accept", func(t *testing.T) {
+		err := cmv.CheckConsensusMessageValidity(cnsMsg, "")
```
Comment (raduchis) [inline consensus/spos/consensusMessageValidator_test.go:537]: should there be an cmv.AddMessageTypeToPublicKey(pubKey, 10, bls.MtBlockBodyAndHeader) here?

File: consensus/spos/constants.go (hunk 1 above)
Comment (iulianpascalau) [reply consensus/spos/constants.go]: done

**Review by AdoAdoAdo — DISMISSED** [review]

**Review by raduchis — DISMISSED** [review]

**Review by SebastianMarian — DISMISSED** [review]

File: consensus/spos/bls/blsWorker.go [hunk 3]
```diff
@@ -17,6 +17,14 @@ import (
 // 2 messages per round (which is ok as it is below the set value of 5)
 const peerMaxMessagesPerSec = uint32(6)
 
+// defaultMaxNumOfMessageTypeAccepted represents the maximum number of the same message type accepted in one round to be
+// received from the same public key for the default message types
+const defaultMaxNumOfMessageTypeAccepted = uint32(1)
+
+// mtSignatureMaxNumOfMessageTypeAccepted represents the maximum number of the signature message type accepted in one round to be
+// received from the same public key
+const mtSignatureMaxNumOfMessageTypeAccepted = uint32(2)
```
Comment (SebastianMarian) [inline consensus/spos/bls/blsWorker.go:26]: maxNumOfMessageTypeSignatureAccepted

File: consensus/spos/bls/blsWorker.go [hunk 4]
```diff
@@ -130,6 +138,15 @@ func (wrk *worker) CanProceed(consensusState *spos.ConsensusState, msgType conse
 	return false
 }
 
+// GetMaxMessageTypeAccepted returns the maximum number of accepted consensus message types per round, per public key
+func (wrk *worker) GetMaxMessageTypeAccepted(msgType consensus.MessageType) uint32 {
```
Comment (SebastianMarian) [inline consensus/spos/bls/blsWorker.go:142]: GetMaxNumOfMessageTypeAccepted

File: consensus/spos/bls/blsWorker_test.go [hunk 5]
```diff
@@ -373,3 +373,23 @@ func TestWorker_IsMessageTypeValid(t *testing.T) {
 	ret = service.IsMessageTypeValid(666)
 	assert.False(t, ret)
 }
+
+func TestWorker_GetMaxMessageTypeAccepted(t *testing.T) {
```
Comment (SebastianMarian) [inline consensus/spos/bls/blsWorker_test.go:377]: GetMaxNumOfMessageTypeAccepted

File: consensus/spos/interface.go [hunk 6]
```diff
@@ -100,6 +100,8 @@ type ConsensusService interface {
 	IsSubroundStartRound(int) bool
 	// GetMaxMessagesInARoundPerPeer returns the maximum number of messages a peer can send per round
 	GetMaxMessagesInARoundPerPeer() uint32
+	// GetMaxMessageTypeAccepted returns the maximum number of accepted consensus message types per round, per public key
+	GetMaxMessageTypeAccepted(msgType consensus.MessageType) uint32
```
Comment (SebastianMarian) [inline consensus/spos/interface.go:104]: GetMaxNumOfMessageTypeAccepted

File: consensus/spos/bls/blsWorker_test.go (hunk 5 above)
Comment (SebastianMarian) [inline consensus/spos/bls/blsWorker_test.go:377]: GetMaxNumOfMessageTypeAccepted

File: consensus/spos/bls/blsWorker.go (hunk 3 above)
Comment (iulianpascalau) [reply consensus/spos/bls/blsWorker.go:26]: done, `maxNumOfMessageTypeSignatureAccepted`

File: consensus/spos/bls/blsWorker.go (hunk 4 above)
Comment (iulianpascalau) [reply consensus/spos/bls/blsWorker.go:142]: renamed

File: consensus/spos/bls/blsWorker_test.go (hunk 5 above)
Comment (iulianpascalau) [reply consensus/spos/bls/blsWorker_test.go:377]: done

**Review by AdoAdoAdo — APPROVED** [review]

**Review by raduchis — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Pull request merged**